src/pj_obs_api.c

```diff
@@ -510,25 +510,198 @@ void proj_context_destroy (PJ_CONTEXT *ctx) {
 }
 
 
-/* Build a fully expanded proj_create() compatible representation of P */
-char *proj_definition_retrieve (PJ *P) {
-    if (0==P)
-        return 0;
-    return pj_get_def(P, 0);
+/*****************************************************************************/
+PJ_INFO proj_info(void) {
+/******************************************************************************
+    Basic info about the current instance of the PROJ.4 library.
+
+    Returns PJ_INFO struct. Searchpath member of the struct is truncated to 512
+    characters.
+
+******************************************************************************/
+    PJ_INFO info;
+    const char **paths;
+    char *tmpstr;
+    int i, n;
+
+    info.major = PROJ_VERSION_MAJOR;
+    info.minor = PROJ_VERSION_MINOR;
+    info.patch = PROJ_VERSION_PATCH;
+    sprintf(info.version, "%d.%d.%d", info.major, info.minor, info.patch);
+    strcpy(info.release, pj_get_release());
+
+    paths = proj_get_searchpath();
+    n = proj_get_path_count();
+    /* build search path string */
+    tmpstr = getenv("HOME");
+    if (tmpstr != NULL) {
+        sprintf(info.searchpath, "%s", tmpstr);
```

Comment:
I don't think it is ANSI C either, so I won't use it. I think I can get around the calls to sprintf in other ways.